neps/nep-0539.md

```diff
@@ -0,0 +1,741 @@
+---
+NEP: 539
+Title: Cross-Shard Congestion Control
+Authors: Waclaw Banasik <waclaw@near.org>, Jakob Meier <inbox@jakobmeier.ch>
+Status: New
+DiscussionsTo: https://github.com/nearprotocol/neps/pull/539
+Type: Protocol
+Version: 1.0.0
+Created: 2024-03-21
+LastUpdated: 2024-04-21
+---
+
+## Summary
+
+We propose to limit the outgoing receipts included in each chunk. Limits depend
+on the congestion level of that receiving shard and are to be enforced by the
+protocol.
+
+Congestion is primarily tracked in terms of the gas of all delayed receipts.
+Chunk producers must ensure they stop accepting new transactions if the receiver
+account is on a shard with a full delayed receipts queue.
+
+Forwarding of receipts that are not directly produced by a transaction, namely
+cross-contract calls and delegated receipts, is limited by the receivers overall
+congestion level. This includes the gas of delayed receipts and the gas of
+receipts that have not been forwarded, yet, due to congestion control
+restrictions. Additionally, the memory consumption of both receipt types can
+also cause congestion.
+
+This proposal replaces the local congestion control rules already in place.
```

Comment:
Nit: I think it supercedes/extends the existing rules rather than fully replacing them (we still have transaction pool limit and delayed receipt queue limit).

Reply:
True, I've added a few words to describe what we keep and what we replace.